Python/tier2_redundancy_eliminator_bytecodes.c

```diff
@@ -189,6 +177,53 @@ dummy_func(void) {
         }
     }
 
+    op(_TO_BOOL, (value -- res)) {
+        if (is_const(value)) {
+            int err = PyObject_IsTrue(get_const(value));
+            ERROR_IF(err < 0, error);
+            OUT_OF_SPACE_IF_NULL(res = sym_new_const(ctx, err ? Py_True : Py_False));
+        }
+        else {
+            OUT_OF_SPACE_IF_NULL(res = sym_new_known_type(ctx, &PyBool_Type));
+        }
+    }
+
+    op(_TO_BOOL_BOOL, (value -- value)) {
+        sym_set_type(value, &PyBool_Type);
+    }
+
+    op(_TO_BOOL_INT, (value -- res)) {
+        sym_set_type(value, &PyLong_Type);
+        if (is_const(value)) {
+            OUT_OF_SPACE_IF_NULL(res = sym_new_const(ctx, _PyLong_IsZero((PyLongObject *)get_const(value))
+                ? Py_False : Py_True));
+        }
+        else {
+            OUT_OF_SPACE_IF_NULL(res = sym_new_known_type(ctx, &PyBool_Type));
+        }
+    }
+
+    op(_TO_BOOL_LIST, (value -- res)) {
+        sym_set_type(value, &PyList_Type);
+        OUT_OF_SPACE_IF_NULL(res = sym_new_known_type(ctx, &PyBool_Type));
+    }
+
+    op(_TO_BOOL_NONE, (value -- res)) {
+        sym_set_type(value, &_PyNone_Type);
+        OUT_OF_SPACE_IF_NULL(res = sym_new_const(ctx, Py_False));
```

Comment:
If `value` is known to be `None` this can be replaced with `_POP_TOP; LOAD_CONST_BORROW`